test/MorseL.Tests/HubWebSocketHandlerTests.cs

```diff
@@ -0,0 +1,164 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using MorseL.Extensions;
+using MorseL.Scaleout;
+using SocketConnection = MorseL.Sockets.Connection;
+using Xunit;
+using MorseL.Sockets;
+
+namespace MorseL.Tests
+{
+    public class HubWebSocketHandlerTests
+    {
+        public class DefaultHub : Hub<IClientInvoker>
+        { }
+
+        public class FailHub : Hub<IClientInvoker>
+        {
+            public override Task OnConnectedAsync(SocketConnection connection)
+            {
+                throw new NotSupportedException();
+            }
+
+            public override Task OnDisconnectedAsync(Exception exception)
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        private readonly Mock<IServiceProvider> ServicesMock = new Mock<IServiceProvider>();
+        private readonly Mock<IServiceScopeFactory> ServiceScopeFactoryMock = new Mock<IServiceScopeFactory>();
+        private readonly Mock<IBackplane> BackplaneMock = new Mock<IBackplane>();
+        private readonly Mock<IOptions<MorseLOptions>> MorseLOptionsMock = new Mock<IOptions<MorseLOptions>>();
+        private readonly Mock<ILoggerFactory> LoggerFactoryMock = new Mock<ILoggerFactory>();
+        private readonly Mock<IChannel> ChannelMock = new Mock<IChannel>();
+        private readonly Mock<IHubActivator<DefaultHub, IClientInvoker>> DefaultHubActivatorMock = new Mock<IHubActivator<DefaultHub, IClientInvoker>>();
+        private readonly Mock<IHubActivator<FailHub, IClientInvoker>> FailHubActivatorMock = new Mock<IHubActivator<FailHub, IClientInvoker>>();
+
+        public HubWebSocketHandlerTests()
+        {
+            ServicesMock.Setup(m => m.GetService(It.Is<Type>(t => t == typeof(IServiceScopeFactory)))).Returns(ServiceScopeFactoryMock.Object);
```

Comment:
I don't think that was this repo? Anyway, that setting never actually prevented concurrent execution /shrug.

Reply:
I mean it seems ideal regardless. I was pondering how best to factor it out since _a lot_ of things need most (all?) of these mocks so having some sort of "mock organizer" object that had these mocks defined seemed good. I'll look at doing that tonight.